conformalInference/R/multisplit.R

```diff
@@ -0,0 +1,193 @@
+#'   R = |Y - m(X)|, and m denotes the estimator produced by train.fun.
+#'   This is used to scale the conformal score, to produce a prediction interval
+#'   with varying local width. The input arguments to mad.train.fun should be
+#'   x: matrix of features, and y: vector of absolute residuals. The default for
+#'   mad.train.fun is NULL, which means that no training is done on the absolute
+#'   residuals, and the usual (unscaled) conformal score is used. Note that if
+#'   mad.train.fun is non-NULL, then so must be mad.predict.fun (see next).
+#' @param mad.predict.fun A function to perform prediction for the (mean of the)
+#'   absolute residuals at new feature values. Its input arguments should be
+#'   out: output produced by mad.train.fun, and newx: feature values at which we
+#'   want to make predictions. The default for mad.predict.fun is NULL, which
+#'   means that no local scaling is done for the conformal score, i.e., the
+#'   usual (unscaled) conformal score is used.
+#' @param split Indices that define the data-split to be used (i.e., the indices
+#'   define the first half of the data-split, on which the model is trained).
+#'   Default is NULL, in which case the split is chosen randomly.
+#' @param seed Integer to be passed to set.seed before defining the random
+#'   data-split to be used. Default is NULL, which effectively sets no seed.
+#'   If both split and seed are passed, the former takes priority and the latter
+#'   is ignored.
+#' @param verbose Should intermediate progress be printed out? Default is FALSE.
+#' @param B number of replications. Default is 50.
+#' @param lambda Smoothing parameter. Default is 0.
+#' @param tau It is a smoothing parameter, whose value affects the behavior of the
+#'  function joining the B intervals:
+#'     tau=1-1/B  Bonferroni intersection method
+#'     tau=0 unadjusted intersection
+#' Default is 1-(B+1)/(2*B).
+#'
+#' @return A list with the following components: lo, up. They are matrices of
+#'  dimension n0 x m. Recall that n0 is the number of rows of x0, and m is the
+#'   number of tuning parameter values internal to predict.fun. In a sense, each
+#'   of the m columns really corresponds to a different prediction function;
+#'   see details below.
+#'
+#' @details For concreteness, suppose that we want to use the predictions from
+#'   forward stepwise regression at steps 1 through 5 in the path. In this case,
+#'   there are m = 5 internal tuning parameter values to predict.fun, in the
+#'   notation used above, and each of the returned matrices pred, lo, up, fit
+#'   will have 5 columns (one for each step of the forward stepwise path).
+#'   The code is structured in this way so that we may defined a single pair of
+#'   functions train.fun and predict.fun, over a set of m = 5 tuning parameter
+#'   values, instead of calling the conformal function separately m = 5 times.
+#'
+#' @details This function is based on the package \code{\link{future.apply}} to
```

Comment:
This can just be a second paragraph of the details section (don't need to write `@details` again).